### PR TITLE
Remove `WorkflowLinter` as it is part of the `Assessment` workflow

### LINKED ISSUE #3035: [FEATURE]: Promote workflow linter to non-experimental
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem statement

The workflow linter is not experimental as it is part of the assessment workflow

### Proposed Solution

Remove the experimental flag from the workflow linter

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by JCZuurmond**
## Changes
Remove `WorkflowLinter` as it is part of the `Assessment` workflow

### Linked issues

Resolves #3035

### Functionality

- [x] removed workflow: `experimental-workflow-linter`

### Tests

- [x] manually tested

### CODE REVIEW

File: src/databricks/labs/ucx/assessment/workflows.py [hunk 1]
```diff
@@ -183,17 +183,21 @@ def crawl_groups(self, ctx: RuntimeContext):
         """Scans all groups for the local group migration scope"""
         ctx.group_manager.snapshot()
 
+    @job_task
+    def assess_workflows(self, ctx: RuntimeContext):
+        """Scans all jobs for migration issues in notebooks jobs.
+
+        Also, stores direct filesystem accesses for display in the migration dashboard.
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/assessment/workflows.py:190]: stores tables as well

Comment (JCZuurmond) [reply src/databricks/labs/ucx/assessment/workflows.py:190]: I am working on a separate PR that updates the docs, separating this out for now

File: README.md [hunk 2]
```diff
@@ -728,9 +728,7 @@ in the Migration dashboard.
 
 ## Jobs Static Code Analysis Workflow
 
-> Please note that this is an experimental workflow.
-
-The `experimental-workflow-linter` workflow lints accessible code from 2 sources:
+The `workflow-linter` workflow lints accessible code from 2 sources:
```
Comment (nfx) [inline README.md:731]: just remove it and modify the assessment documentation. it's part of assessment and migration progress

File: src/databricks/labs/ucx/assessment/workflows.py [hunk 3]
```diff
@@ -183,17 +183,21 @@ def crawl_groups(self, ctx: RuntimeContext):
...
+        """
+        ctx.workflow_linter.refresh_report(ctx.sql_backend, ctx.inventory_database)
+
     @job_task
     def assess_dashboards(self, ctx: RuntimeContext):
         """Scans all dashboards for migration issues in SQL code of embedded widgets.
-        Also stores direct filesystem accesses for display in the migration dashboard."""
```
Comment (nfx) [inline src/databricks/labs/ucx/assessment/workflows.py:189]: what is this diff about?

**Review by nfx — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (JCZuurmond) [reply README.md:731]: done

**Review by asnare — COMMENTED** [review]
This workflow is also mentioned in the `docs/table_persistance.md` table, which can be cleaned up.

**Review by asnare — APPROVED** [review]
lgtm

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**